src/session_pool/session_pool.js

```diff
@@ -0,0 +1,312 @@
+import EventEmitter from 'events';
+import log from 'apify-shared/log';
+import { checkParamOrThrow } from 'apify-client/build/utils';
+
+import { openKeyValueStore } from '../key_value_store';
+import { Session } from './session';
+import events from '../events';
+import { ACTOR_EVENT_NAMES_EX } from '../constants';
+
+
+/**
+ * Handles the sessions rotation, creation and persistence.
```

Comment:
I'd add a note that this is Experimental feature and might change in the future releases.